pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegment.java

```diff
@@ -32,10 +33,40 @@
    */
   boolean index(GenericRow row);
 
+  /**
+   * Indexes a record into the segment with optionally provided metadata.
+   *
+   * @param row Record represented as a {@link GenericRow}
+   * @param msgMetadata the metadata associated with the message
+   * @return Whether the segment is full (i.e. cannot index more record into it)
+   */
+  default boolean index(GenericRow row, StreamMessageMetadata msgMetadata) {
```

Comment:
This interface is not meant to be used by any Stream implementation (or other injected implementations). We can go ahead and modify the index() method in line 34.

Reply:
Agreed. I was being defensive. 

I think we should annotate the interfaces as "public" and "internal" so when we are updating or reviewing updates we know when to look for backward compatibility.